app/assets/src/api/index.js

```diff
@@ -251,7 +251,56 @@ const createProject = params =>
 const logAnalyticsEvent = (eventName, eventData = {}) => {
   // Wrapper around Segment analytics so we can add things later
   // eventData should have keys in snake_case for the database
-  if (window.analytics) window.analytics.track(eventName, eventData);
+  if (window.analytics) {
+    // Include high value user groups in event properties to avoid JOINs downstream.
+    if (window.analytics.user) {
+      const traits = window.analytics.user().traits();
+      eventData = {
+        // see traits_for_segment
+        admin: traits.admin,
+        biohub_user: traits.biohub_user,
+        czi_user: traits.biohub_user,
+        demo_user: traits.demo_user,
+        has_samples: traits.has_samples,
+        // caller can override above traits if they know what they are doing
+        ...eventData
+      };
+    }
+    window.analytics.track(eventName, eventData);
+  }
+};
+
+/**
+ * For wrapping event handlers in React. The first part of the name should be
+ * the React component. The middle part should be user-friendly UI name. The
+ * last part should be the action taken.
+ *
+ * <ReactComponent>_<friendly_name_of_ui_element>_<past_tense_action>
+ *
+ * In this way, we can easily discern the
+ * context and meaning of an analytics event in a report and locate it in the
+ * codebase.
+ *
+ * For example:
+ *
+ *    withAnalytics(
+ *      this.renderMoreReads,
+ *      "AccessionViz_more_reads_link_clicked",
+ *      { reads: this.state.reads.length, allReads: this.allReads.length }
+ *    )
+ *
+ * React events should have have a single callsite, so there is no need to put
+ * them in ANALYTICS_EVENT_NAMES.
+ *
+ * The eventData should be used for extra data that would be useful for an
+ * analysis specific to the event.
+ **/
+const withAnalytics = (handleEvent, eventName, eventData = {}) => {
+  return (...args) => {
+    const ret = handleEvent.apply(null, args);
```

Comment:
Can you just do `handleEvent(...args)` here? 

`apply` will also set `this` to `null`, which I don't think is intended?

Reply:
good catch!